File: 0_Revision/twoSum.js
const nums = [2, 7, 11, 15];
const target = 9;


function twoSum(nums, target) {
    const hashMap = {};
    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i];
        if (hashMap[complement] !== undefined) {
            return [hashMap[complement], i]
        }
        hashMap[nums[i]] = i;
    }
}

console.log(twoSum(nums, target));

File: 0_Revision/twoSum.py
nums = [2, 7, 11, 15]

target = 9

def twoSum(nums, target):
    hashMap = {}
    for i, num in enumerate(nums):
        complement = target - num
        if complement in hashMap:
            return [hashMap[complement], i]
        hashMap[num] = i

print(twoSum(nums, target))